env0/data_agents.go

```diff
@@ -0,0 +1,54 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataAgents() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: agentsRead,
+
+		Schema: map[string]*schema.Schema{
+			"agents": {
+				Type:        schema.TypeList,
+				Description: "list of organization agents",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"agent_key": {
+							Type:        schema.TypeString,
+							Description: "agent key identifier",
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func agentsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	agents, err := apiClient.Agents()
+	if err != nil {
+		return diag.Errorf("could not read agents: %v", err)
+	}
+
+	data := struct {
+		Agents []client.Agent
+	}{agents}
+
+	if err := writeResourceData(&data, d); err != nil {
+		return diag.Errorf("schema resource data serialization failed: %v", err)
+	}
+
+	// Not really needed. But required by Terraform SDK - https://github.com/hashicorp/terraform-plugin-sdk/issues/541
+	d.SetId("1")
```

Comment:
Does that mean that a user cannot have more than a single data resource of agents in his TF code?

Reply:
@yaronya ,

I can make it random if preferred.
However, why would using "1" as the ID prevent it from being used in multiple places? 

See this bug fix in the google provider:
https://github.com/hashicorp/terraform-provider-google/pull/7375/files

Reply:
> However, why would using "1" as the ID prevent it from being used in multiple places?

I think that the id has to be unique across the TF state, isn't it? 
Random is also not so great, but preferable than `"1"`.
I'd try and see if we can get from the schema/context the resource name and set it based on it (since it has to be unique anyways on your TF stack).
WDYT?

> See this bug fix in the google provider

The example you just sent has the id based on some dynamic parameters such as project id and zone id

Reply:
@yaronya - regardless wouldn't it potentially cause duplication of ID (the example I sent).

Reply:
> @yaronya - regardless wouldn't it potentially cause duplication of ID (the example I sent).

@TomerHeber 
Yes, I believe it may create such duplication, but that example has somewhat lower chance to do so as it depends on **at least some** parameters
I'm assuming it'd break stuff as the id has to be unique, but maybe I'm wrong.